src/main/java/crawlercommons/fetcher/http/UserAgent.java

```diff
@@ -17,22 +17,34 @@
 package crawlercommons.fetcher.http;
 
 import java.io.Serializable;
-import java.util.Locale;
 
 import crawlercommons.util.HttpFetcherVersion;
 
 /**
  * User Agent enables us to describe characteristics of any http-fetcher
- * agent. There are a number of constructor options to describe the following:
+ * agent. There are a number of configurable options to describe the following:
  * <ol>
- * <li><code>_agentName</code>: Primary agent name.</li>
- * <li><code>_emailAddress</code>: The agent owners email address.</li>
- * <li><code>_webAddress</code>: A web site/address representing the agent owner.</li>
- * <li><code>_browserVersion</code>: Broswer version used for compatibility.</li>
- * <li><code>_crawlerVersion</code>: Version of the user agents personal crawler. If
+ * <li><code>agentName</code>: Primary agent name.</li>
+ * <li><code>emailAddress</code>: The agent owners email address.</li>
+ * <li><code>webAddress</code>: A web site/address representing the agent owner.</li>
+ * <li><code>browserVersion</code>: Browser version used for compatibility.</li>
+ * <li><code>crawlerVersion</code>: Version of the user agents personal crawler. If
  * this is not set, it defaults to the http-fetcher maven artifact version.</li>
+ * <li><code>userAgentString</code>: (Optional) the user-agent string is constructed
+ * automatically if not provided, but can be fully customized via this option.</li>
  * </ol>
- * 
+ * <p>
+ * To construct an UserAgent, you can use a <code>UserAgent.Builder</code> object:
+ * <pre>
+ * {@code
+ *     UserAgent ua = new UserAgent.Builder()
+ *             .setAgentName("MyCrawler")
+ *             .setCrawlerVersion("1.0")
+ *             .setWebAddress("www.mycrawler.com/bot.html")
+ *             .setUserAgentString("Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P)")
```

Comment:
Is there a specific use case to set agentName, etc. and then the entire user-agent string? Wouldn't the parts set beforehand just overwritten?

Reply:
In the long term it might be worth to consider whether a routine to fill the HTTP User-Agent header could be part of the crawler-commons library.

Reply:
Indeed, I don't see that use case.  The main motivation here was just allowing the user to override the full user-agent string, while still allowing to auto-generating it from agentName/crawlerVersion.

I'm not sure I understood the second comment. Is it about moving this feature to the main crawler-commons library?

Reply:
>  Is it about moving this feature to the main crawler-commons library?

Yes. Just a thought: putting together the user-agent string, is something every crawler implements.

Reply:
Yeah, makes sense. 
Thanks, for the review.